### DISCUSSION TITLE
Missing parts of the database

### DISCUSSION THREAD

**Discussion by TenebrisWoW** [azerothcore/azerothcore-wotlk#10137] [General]
Hello dear community! I hereby ask for your support again, because I have the following problem:

Large parts of the spell_dbc are missing from my database. I tried to extract it in order to import it, but I can't and can't handle the tools.

**I would be very grateful if someone could provide me with their spell_dbc. You can also reach me on Discord as "Qu | Michael # 3897"!**

I wouldn't ask here if there was any other way to fix it. I have worked my way into a lot, but unfortunately I still can't get some things right after a whole day of work. :-)

In any case, I would be very grateful if someone could bring themselves to help.

**Comment by Nyeriah** [comment]
That table is supposed to be mostly empty. It's only used by a few spell corrections and custom stuff.

**Comment by TenebrisWoW** [comment]
But if I want to edit a spell, I can only do so if it is also in the database. For example, I want to "ride" the required level for the spell. And many others. So I have a biggest problem.

Very few players will be playing on the server. For example, it makes no sense if a transport with 3 people has to be canalized at a meeting stone.

**Comment by TenebrisWoW** [comment]
This is my error message when I start the "install npm" command via Power Shell. No matter where I unzip the folder of the "node_dbc_reader".

**PS D:\nodejs> npm install
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path D:\nodejs/package.json
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, open 'D:\nodejs\package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent**

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\admin\AppData\Local\npm-cache\_logs\2022-01-11T23_21_05_341Z-debug.log

**Reply by TenebrisWoW** [reply]
Or:

**PS D:\nodejs> npm run start -- --search="{*} == 100" --columns=33388 Spell
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path D:\nodejs/package.json
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, open 'D:\nodejs\package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! eno**

**Comment by TenebrisWoW** [comment]
I am now a few steps further. I understand that I don't have to change the spells in the database, but in the DBC file. In most cases, a patch would then also be necessary for the client, but there are also some entries that should be changeable in this way. For example the casting times of skinning, mining and herbalism. Or the cast time of the Hearthstone. I used the DBC tool from TRX, which I took from a German video. However, if I only make the slightest change, the Wordserver no longer starts and the following error message appears:

**D:\World of Warcraft Privatserver\Source\src\common\DataStores\DBCFileLoader.h:75 in DBCFileLoader::Record::getString ASSERTION FAILED:
  stringOffset < file.stringSize**

I also noticed that the size of the file has decreased from 47800 KB to around 45000 KB and even if I undo the changes, the world server starts with the same error.

Does anyone have an idea or another program that I can use under Windows?
